CHANGELOG.md

```diff
@@ -1,5 +1,106 @@
 # Changelog for Elixir v1.7
 
+Elixir v1.7 is the last release to support Erlang/OTP 19. We recommend everyone to migrate to Erlang/OTP 20+.
+
+## The `__STACKTRACE__` construct
+
+Erlang/OTP 21.0 introduces a new way to retrieve the stacktrace that is lexically scoped and no longer relies on side-effects like `System.stacktrace/0` does. Before one would write:
+
+```elixir
+try do
+  ... something that may fail ...
+rescue
+  e ->
+    log(e, System.stacktrace())
+    reraise e, System.stacktrace()
+end
+```
+
+In Elixir v1.7, this can be written as:
+
+```elixir
+try do
+  ... something that may fail ...
+rescue
+  e ->
+    log(e, __STACKTRACE__)
+    reraise e, __STACKTRACE__
+end
+```
+
+This change may also yield performance improvements in the future, since the lexical scope allows us to track precisely when a stacktrace is used and we no longer need to keep references to stacktrace entries after the `try` construct finishes.
+
+Other parts of the exception system have been improved. For example, more information is provided on error messages certain occurrences of `ArgumentError`, `ArithmeticError` and `KeyError`.
+
+## Erlang/OTP logger integration
+
+Erlang/OTP 21 includes a new `:logger` module. Elixir v1.7 fully integrates with the new `:logger` and leverages its metadata system. The `Logger.Translator` mechanism has also been improved to export metadata, allowing custom Logger backends to leverage information such as:
+
+  * `:crash_reason` - a two-element tuple with the throw/error/exit reason as first argument and the stacktrace as second
+
+  * `:initial_call` - the initial call that started the process
+
+  * `:registered_name` - the process registered name as an atom
+
+We recommend Elixir libraries that previously hooked into Erlang's `:error_logger` to hook into `Logger` instead, so those libraries support all current and future Erlang/OTP versions.
```

Comment:
"... to hook into `Logger` instead ~so those libraries~ _in order to_ support all ..."